project-koku/koku
# [COST-4125] - adding usecols to AWS downloader flow

## Jira Ticket

[COST-4125](https://issues.redhat.com/browse/COST-4125)

## Description

This change will add usecols to the AWS downloader flow to filter on columns in our recommended list, shaving off unnecessary columns and also saving us some memory.

## Testing

1. Checkout Branch
2. Restart Koku
3. Make create-test-customer
4. Make load-test-customer-data
5. Everything should process normally

Optionally add a bonus column that DOES NOT match anything in your dataset to `RECOMMENDED_COLUMNS`
Rerun the above and see we continue to process the data even if there is a column missing.

## Notes
Locally this change took our large 10G data set down to 2G

Bogus column example
`{'message': 'pandas read csv with following usecols', 'tracing_id': '20230823T030139Z', 'provider_uuid': '94261544-19ae-491e-b714-ab6f08074cdf', 'provider_type': 'AWS', 'account': 'org1234567', 'usecols': {'BOGUS COL', 'bill/InvoiceId', .................}}`

This `BOGUS COL` was not in my dataset but we processed the data fine

`copy_local_report_file_to_s3_bucket: data/csv/org1234567/AWS/source=94261544-19ae-491e-b714-ab6f08074cdf/year=2023/month=08 /testing/data/processing/org1234567/aws/redhat-cost-management-bucket-0bddd459765c3f6b/2023-08-09_0_0.csv`
...
